python/ray/tune/execution/trial_runner.py

```diff
@@ -124,6 +130,7 @@ def __init__(
         sync_trial_checkpoints: bool,
         local_dir: str,
         remote_dir: str,
+        trial_checkpoint_config: Optional[CheckpointConfig] = None,
```

Comment:
It seems we pass the trial checkpoint config only to retrieve the num_to_keep.

Can we instead consider the two options:

1. Only pass trial_num_to_keep (or sync_every_n_trial_checkpoints or so)
2. Use the trial_checkpoint config also to replace the `syncer`, `sync_trial_checkpoints`, and `remote_dir` attribute? (only if it makes sense/simplifies the code)